### PR TITLE
Add debug CLI and wrap errors

### PR DESCRIPTION
**Pull request by guggero**
Adds a debug CLI that helps inspecting a local trader database. Also wraps some errors to give more information about the cause.

EDIT: The advantage of these commands is that they work without starting the trader daemon (so no interaction with the server) and also work on databases of other users, as no interaction with lnd is required either.
Perhaps it's really just something we'll use and it doesn't need to be in the main code base. But I've hidden the command group so most people won't notice this.

### CODE REVIEW

File: auctioneer/batch.go [hunk 1]
```diff
@@ -67,15 +67,15 @@ func (c *Client) checkPendingBatch() error {
 // finalizedBatchTx retrieves the finalized transaction of a batch according to
 // the auctioneer, i.e., the transaction that will be broadcast to the network.
 func (c *Client) finalizedBatchTx(id order.BatchID) (*wire.MsgTx, error) {
-	req := &poolrpc.RelevantBatchRequest{Id: id[:]}
-	batch, err := c.client.RelevantBatchSnapshot(context.Background(), req)
+	req := &poolrpc.BatchSnapshotRequest{BatchId: id[:]}
```
Comment (halseth) [inline auctioneer/batch.go:70]: Hm, should we just remove the `RelevantBatchRequest` RPC?

**Review by halseth — CHANGES_REQUESTED** [review]
I think it is better to split off the debug command :)

**Comment by guggero** [comment]
Split the actual bug fix off into https://github.com/lightninglabs/pool/pull/179 and removed the `RelevantBatchSnapshot` RPC there.

File: cmd/pool/debug.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+var debugCommands = []cli.Command{
+	{
+		Name:      "debug",
+		ShortName: "d",
+		Usage:     "Debug the local trader state.",
+		Hidden:    true,
+		Subcommands: []cli.Command{
+			dumpAccountsCommand,
+			dumpOrdersCommand,
```
Comment (wpaulino) [inline cmd/pool/debug.go:27]: Any reason to have these given that we have `pool accounts list` and `pool orders list`?

**Review by wpaulino — APPROVED** [review]

File: cmd/pool/debug.go (hunk 2 above)
Comment (guggero) [reply cmd/pool/debug.go:27]: I should have given some more explanation, you're right. I did this in a hurry to get a fix for the bug out quickly.
The advantage of these commands is that they work without starting the trader daemon (so no interaction with the server) and also work on databases of other users, as no interaction with `lnd` is required either.
Perhaps it's really just something we'll use and it doesn't need to be in the main code base. But I've hidden the command group so most people won't notice this.

**Review by halseth — APPROVED** [review]
Looks useful, LGTM :)

**Pull request merged**